test/swar/BasicOperations.cpp

```diff
@@ -86,31 +86,29 @@ ARRAY_TEST(S8_64, 255, 255, 255, 255, 255, 255, 255, 255);
 ARRAY_TEST(S16_32, 65534, 65534);
 ARRAY_TEST(S16_64, 65534, 65534, 65534, 65534);
 
-#define F false
-#define T true
 using BS = BooleanSWAR<4, u16>;
-static_assert(BS{Literals<4, u16>, {F, F, F, F}}.value() == 0b0000'0000'0000'0000);
-static_assert(BS{Literals<4, u16>, {T, F, F, F}}.value() == 0b1000'0000'0000'0000);
-static_assert(BS{Literals<4, u16>, {F, T, F, F}}.value() == 0b0000'1000'0000'0000);
-static_assert(BS{Literals<4, u16>, {F, F, T, F}}.value() == 0b0000'0000'1000'0000);
-static_assert(BS{Literals<4, u16>, {F, F, F, T}}.value() == 0b0000'0000'0000'1000);
-static_assert(BS{Literals<4, u16>, {T, F, F, F}}.value() == 0b1000'0000'0000'0000);
+static_assert(BS{Literals<4, u16>, {0, 0, 0, 0}}.value() == 0b0000'0000'0000'0000);
+static_assert(BS{Literals<4, u16>, {1, 0, 0, 0}}.value() == 0b1000'0000'0000'0000);
+static_assert(BS{Literals<4, u16>, {0, 1, 0, 0}}.value() == 0b0000'1000'0000'0000);
+static_assert(BS{Literals<4, u16>, {0, 0, 1, 0}}.value() == 0b0000'0000'1000'0000);
+static_assert(BS{Literals<4, u16>, {0, 0, 0, 1}}.value() == 0b0000'0000'0000'1000);
+static_assert(BS{Literals<4, u16>, {1, 0, 0, 0}}.value() == 0b1000'0000'0000'0000);
 
 
 namespace equality {
 using S = SWAR<8, u32>;
 using BS = BooleanSWAR<8, u32>;
-static_assert(equals(S{S::Literal,      {1, 2, 3, 4}},
-                     S{S::Literal,      {1, 2, 3, 4}}).value()
-                     == BS{BS::Literal, {T, T, T, T}}.value());
-
-static_assert(equals(S{S::Literal,      {1, 2, 3, 4}},
-                     S{S::Literal,      {5, 6, 7, 8}}).value()
-                     == BS{BS::Literal, {F, F, F, F}}.value());
-
-static_assert(equals(S{S::Literal,      {1, 2, 3, 4}},
-                     S{S::Literal,      {5, 2, 7, 4}}).value()
-                     == BS{BS::Literal, {F, T, F, T}}.value());
+template <typename S = S, typename BS = BS>
+constexpr auto laneWiseEqualsTest(
```

Comment:
This is not right, if you want "all lanes equal" you use the underlying value comparison as in all other tests.
Unit tests ought to demonstrate how to use a codebase.  Using per-lane comparisons for a whole-base-type comparison is inadequate.
There is a probem with `equals` which is that is terseness invites incorrect use.
Even if you use `equals`, the comparison against a `BooleanSWAR` of all `true` is inelegant:
`not(boolean(~equals(`, since we have the conversion to boolean